Add TC2030 pkg/dev (with and without legs)

Tubbles opened the pull request:
##### SUMMARY

Added Tag-Connect TC2030 6-pin Plug-of-Nails™ connector package with some variants

https://www.tag-connect.com/product/tc2030-idc-6-pin-tag-connect-plug-of-nails-spring-pin-cable-with-legs

![image](https://github.com/LibrePCB-Libraries/Tag_Connect.lplib/assets/2370561/1405e33f-7a20-44da-9fd8-0e27cce7b896)

https://www.tag-connect.com/product/tc2030-idc-nl

![image](https://github.com/LibrePCB-Libraries/Tag_Connect.lplib/assets/2370561/5c4bc2a0-db45-4741-9874-d1827f154251)

https://www.tag-connect.com/product/tc2030-retaining-clip-board-3-pack

![image](https://github.com/LibrePCB-Libraries/Tag_Connect.lplib/assets/2370561/1a1de944-ec62-44f1-a8e0-bfd941e41e54)


##### DATASHEETS / REFERENCES

https://www.tag-connect.com/wp-content/uploads/bsk-pdf-manager/2019/12/TC2030-IDC-Datasheet-Rev-B.pdf
https://www.tag-connect.com/wp-content/uploads/bsk-pdf-manager/TC2030-CLIP_Datasheet_24.pdf

##### OPEN QUESTIONS / UNRESOLVED ISSUES

Unclear usage of layers, particularly top/bot courtyard. Maybe I've missed to use some layers?

##### CHECKLIST

- [x] I have read and followed the library conventions¹.
- [ ] ~~For packages, I followed IPC7351C (see details in library conventions).~~
- [ ] ~~For modified elements, I bumped their version number².~~
- [ ] I already used the new library elements in a design and verified their correctness (optional).
- [x] I'm the copyright owner of the added content (i.e. the changes are made by myself, not copied/imported from somewhere else).
- [x] I agree to publish all my changes under the CC0 Public Domain License³, allowing everyone to use and modify the content without any restrictions.

¹ *Library Conventions: https://docs.librepcb.org/#libraryconventions*
² *Minor version bump if only metadata was modified (e.g. "0.1" -> "0.1.1"), major version bump if functional changes were made (e.g. "0.1" -> "0.2")*
³ *CC0 Public Domain License: https://en.wikipedia.org/wiki/CC0*

Tubbles committed:
Add TAGCONNECT_TC2030 package, variants: default, no leg, no leg with clip

ubruhin commented:
Awesome, thank you very much! I'm sure sooner or later I'll need this connector in a private project :slightly_smiling_face: 

Overall it looks very good, but (as always :see_no_evil:) I have some feedback:

- I'd rotate the footprints by 90° CW for consistency with the "pin-1 at top left" convention followed by most other footprints. The name/value texts can be left at their current position, the smaller distance to the footprint outline is fine.
- The keep-out polygon could be moved to the "Comments" layer as well to make clear it is related to the comment. Feel free to use thinner lines, it might look a bit "lighter".
- In the "default" footprint, there are two overlapping lines on the "Top Documentation" layer. It's not critical, but usually it looks nicer if the overlapping polygon segment is removed (i.e. convert the rectangle to a U-shape).
- There is currently no polygon on the silkscreen ("Top Placement" layer). I think it would be nice to somehow see the connector location on silkscreen. Maybe draw the outline of the connector head on silkscreen, i.e. a line just around the pads (as drawn in the datasheet)?
- I wonder if it would make more sense to have different packages (and devices) for the "Legged" and "No Legs" connectors :thinking: In the end, these are two different connectors and it *might* be desired to choose the variant already in the schematic rather than in the board (devices can be chosen in the schematic, but footprints not). Would you agree to split it into two packages? Of course "clip" vs. "no clip" makes sense as footprint variants since both footprints correspond to the same connector.

Tubbles committed:
Fix review comments on packages PR to TC2030

Tubbles committed:
Add corresponding devices for TC2030

Tubbles commented:
>     * I'd rotate the footprints by 90° CW for consistency with the "pin-1 at top left" convention followed by most other footprints. The name/value texts can be left at their current position, the smaller distance to the footprint outline is fine.

Done!

>     * The keep-out polygon could be moved to the "Comments" layer as well to make clear it is related to the comment. Feel free to use thinner lines, it might look a bit "lighter".

Done!

>     * In the "default" footprint, there are two overlapping lines on the "Top Documentation" layer. It's not critical, but usually it looks nicer if the overlapping polygon segment is removed (i.e. convert the rectangle to a U-shape).

Done!

>     * There is currently no polygon on the silkscreen ("Top Placement" layer). I think it would be nice to somehow see the connector location on silkscreen. Maybe draw the outline of the connector head on silkscreen, i.e. a line just around the pads (as drawn in the datasheet)?

Done!

>     * I wonder if it would make more sense to have different packages (and devices) for the "Legged" and "No Legs" connectors 🤔 In the end, these are two different connectors and it _might_ be desired to choose the variant already in the schematic rather than in the board (devices can be chosen in the schematic, but footprints not). Would you agree to split it into two packages? Of course "clip" vs. "no clip" makes sense as footprint variants since both footprints correspond to the same connector.

Yeah that makes sense to me, the legged variant is more suitable for development and debugging activities, while the no legged variant is more suitable for series board models and temporary connection for firmware flashing. I've split them up

Also added corresponding devices :D

ubruhin committed:
Apply some improvements to TC2030

ubruhin commented:
I allowed myself to rebase this branch, upgrade it to the 1.0 file format and apply some small improvements :slightly_smiling_face: 

![image](https://github.com/LibrePCB-Libraries/Tag_Connect.lplib/assets/5374821/73ac3865-49a8-460c-b49f-90faa8e3faa5)

@Tubbles Note that both devices linked to the same package. I fixed that but assigned a new UUID to the modified device to avoid breaking anything on your side if you already used that device.

I'll merge this now. If you still see some issue, please let me know.
